tests/test_crawl.py

```diff
@@ -281,3 +287,57 @@ def test_main(mocked_launch, pipe_writer):
     # This ensures that pipe is writable even if main program is fininshed -
     # e.g. for threads that are not closed yet.
     assert not pipe_writer.close.called
+
+
+def test_image_info(tmp_path):
+    project_dir = create_project(tmp_path)
+    out, err = call_command(project_dir, "shub-image-info")
+    # can't be asserted as it contains a SHScrapyDeprecationWarning
+    # assert err == ""
+    data = json.loads(out)
+    expected = {
+        "project_type": "scrapy",
+        "spiders": ["myspider"],
+        "metadata": {"myspider": {}},
+    }
+    if not SPIDER_METADATA_AVAILABLE:
+        del expected["metadata"]
+    assert data == expected
+
+
+@pytest.mark.skipif(not SPIDER_METADATA_AVAILABLE, reason="scrapy-spider-metadata is not installed")
```

Comment:
It seems to make more sense to check the output in both cases, not skip the test.